flume-ng-sinks/flume-hdfs-sink/src/test/java/org/apache/flume/sink/hdfs/TestHDFSEventSink.java

```diff
@@ -1323,6 +1323,64 @@ public void testCloseOnIdle() throws IOException, EventDeliveryException, Interr
     fs.close();
   }
 
+  @Test
+  public void testBlockCompressSequenceFileWriterSync() throws IOException, EventDeliveryException, InterruptedException {
+    String hdfsPath = testPath + "/sequenceFileWriterSync";
+    String body = "test event-" + System.nanoTime();
+
+    Configuration conf = new Configuration();
+    FileSystem fs = FileSystem.get(conf);
+
+    // Since we are reading a partial file we don't want to use checksums
+    fs.setVerifyChecksum(false);
+    fs.setWriteChecksum(false);
+
+    Path dirPath = new Path(hdfsPath);
+    fs.delete(dirPath, true);
+    fs.mkdirs(dirPath);
+
+    Context context = new Context();
+    context.put("hdfs.path", hdfsPath);
+    // Ensure the file isn't closed and rolled, we are only going to send one event
+    context.put("hdfs.rollCount", "10");
+    context.put("hdfs.rollSize", "0");
+    context.put("hdfs.rollInterval", "0");
+    context.put("hdfs.batchSize", "1");
+    context.put("hdfs.fileType", "SequenceFile");
+    context.put("hdfs.codeC", "BZip2Codec"); // Compression codec that doesn't require native hadoop libraries
+    context.put("hdfs.writeFormat", "Writable");
+    Configurables.configure(sink, context);
+
+    Channel channel = new MemoryChannel();
+    Configurables.configure(channel, context);
+
+    sink.setChannel(channel);
+    sink.start();
+
+    Transaction txn = channel.getTransaction();
+    txn.begin();
+    Event event = new SimpleEvent();
+    event.setBody(body.getBytes());
+    channel.put(event);
```

Comment:
>> Test all of the following codecs: bzip2, gzip, snappy
> Snappy and gzip require the native hadoop libraries. Do you know of a work around here to enable that?
I believe there are pure Java versions of both snappy and gzip compression. The native libraries are optional. For example, snappy-java is a pure Java implementation of snappy, and I think it ships with Hadoop. Please let me know if this doesn't appear to be working for you.

>  I thought that writing one event (and thus a corresponding batch size of one) would suffice. Is there a scenario that I'm missing where multiple events would be different?
I want to make sure that multiple batches concatenated is still readable in all cases. I suppose more than one event per batch doesn't add much coverage, but certainly we need to test multiple batches in order to ensure that the SequenceFile framing is working properly across multiple sync() calls on the same file.